enumerables.rb

```diff
@@ -40,12 +75,24 @@ def my_map(&block)
       map_arr.push(block.call(i))
     end
     map_arr
+    to_enum
   end
```

Comment:
- [ ]  enumerables #my_map returns a new array with the results of running a given block once for every element in enum.
      Failure/Error: expect(array.my_map { |num| num < 10 }).to eq(array.map { |num| num < 10 })

- [ ] enumerables #my_map returns a new array with the results of running a given block once for every element in enum.
      Failure/Error: let(:block) { proc { |num| num < (LOWEST_VALUE + HIGHEST_VALUE) / 2 } }

- [ ] enumerables #my_map returns a new array with the results of running a given block once for every element in enum::range
      Failure/Error: expect(range.my_map(&block)).to eq range.map(&block)

- [ ]  enumerables #my_map returns an Enumerator if no block is given
      Failure/Error: expect(array.my_map).to be_an(Enumerator)

- [ ]  enumerables #my_map accepts a proc as an argument and returns a new array with the results of running the proc once for every element in enum.
      Failure/Error: let(:my_proc) { proc { |num| num > 10 } }

- [ ] enumerables #my_map executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.
      Failure/Error: expect(array.my_map(my_proc) { |num| num < 10 }).to eq array.map(&my_proc)

- [ ] enumerables #my_map does not mutate the original array
      Failure/Error: array.my_map { |num| num + 1 }